test/service/ratelimit_test.go

```diff
@@ -240,3 +240,40 @@ func TestInitialLoadError(test *testing.T) {
 	t.assert.Equal("no rate limit configuration loaded", err.Error())
 	t.assert.EqualValues(1, t.statStore.NewCounter("call.should_rate_limit.service_error").Value())
 }
+
+func TestUnlimited(test *testing.T) {
+	t := commonSetup(test)
+	defer t.controller.Finish()
+	service := t.setupBasicService()
+
+	request := common.NewRateLimitRequest(
+		"some-domain", [][][2]string{{{"foo", "bar"}}, {{"hello", "world"}}, {{"baz", "qux"}}}, 1)
+	limits := []*config.RateLimit{
+		config.NewRateLimit(10, pb.RateLimitResponse_RateLimit_MINUTE, t.statsManager.NewStats("foo_bar"), false),
+		nil,
+		config.NewRateLimit(55, pb.RateLimitResponse_RateLimit_SECOND, t.statsManager.NewStats("baz_qux"), true)}
+	t.config.EXPECT().GetLimit(nil, "some-domain", request.Descriptors[0]).Return(limits[0])
+	t.config.EXPECT().GetLimit(nil, "some-domain", request.Descriptors[1]).Return(limits[1])
+	t.config.EXPECT().GetLimit(nil, "some-domain", request.Descriptors[2]).Return(limits[2])
+
+	expectedCacheLimits := []*config.RateLimit{limits[0], nil, nil}
+
+	t.cache.EXPECT().DoLimit(nil, request, expectedCacheLimits).Return([]*pb.RateLimitResponse_DescriptorStatus{
+		{Code: pb.RateLimitResponse_OK, CurrentLimit: limits[0].Limit, LimitRemaining: 9},
+		{Code: pb.RateLimitResponse_OK, CurrentLimit: nil, LimitRemaining: 0},
+		{Code: pb.RateLimitResponse_OK, CurrentLimit: nil, LimitRemaining: 0},
+	})
+
+	response, err := service.ShouldRateLimit(nil, request)
+	common.AssertProtoEqual(
+		t.assert,
+		&pb.RateLimitResponse{
+			OverallCode: pb.RateLimitResponse_OK,
+			Statuses: []*pb.RateLimitResponse_DescriptorStatus{
+				{Code: pb.RateLimitResponse_OK, CurrentLimit: limits[0].Limit, LimitRemaining: 9},
+				{Code: pb.RateLimitResponse_OK, CurrentLimit: nil, LimitRemaining: 0},
+				{Code: pb.RateLimitResponse_OK, CurrentLimit: nil, LimitRemaining: 0},
```

Comment:
@mattklein123 this is something I'm concerned with. We cannot distinguish between a descriptor that wasn't found and a descriptor that's unlimited :(. This will make it hard for the clients to decide what to do. Any ideas how we can fix this without changing the proto spec?

Reply:
Yeah good point. It's not perfect but in the unlimited case why don't we set LimitRemaining to max uint? We can document that. In a perfect world we would add a boolean to the proto but I'm not sure it's worth it right now?

Reply:
Yea that sounds good to me, changed to max.uint32_t